### PR TITLE
Limit metric tag to concrete message type

### PR DESCRIPTION
**Pull request by lailabougria**
_related to https://github.com/Particular/NServiceBus/issues/7022_

Currently, a metric tag containing the content of the EnclosedMessageTypes header is added to the Fetches, Failures, and Success metrics emitted through System.Diagnostics in Core.

However, the value of this header can be quite large, and doesn't provide as much value, as filtering makes most sense on the message type, not its entire hierarchy tree.
This PR changes that value to be limited to the most concrete message type, and only includes the message type full name, and excludes assembly info.

The full contents of the header are still emitted as trace tags.

### CODE REVIEW

File: src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs [hunk 1]
```diff
@@ -18,12 +19,14 @@ public ReceiveDiagnosticsBehavior(string queueNameBase, string discriminator)
     public async Task Invoke(IIncomingPhysicalMessageContext context, Func<IIncomingPhysicalMessageContext, Task> next)
     {
         context.MessageHeaders.TryGetValue(Headers.EnclosedMessageTypes, out var messageTypes);
+        string messageTypeHeader = !string.IsNullOrEmpty(messageTypes) ? messageTypes.Split(';').FirstOrDefault() : default;
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs:22]:
```suggestion
        var messageTypeHeader = !string.IsNullOrEmpty(messageTypes) ? messageTypes.Split(';').FirstOrDefault() : default;
```

File: src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs [hunk 2]
```diff
@@ -18,12 +19,14 @@ public ReceiveDiagnosticsBehavior(string queueNameBase, string discriminator)
...
+        string messageTypeName = !string.IsNullOrEmpty(messageTypeHeader) ? messageTypeHeader.Split(',').FirstOrDefault() : default;
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs:23]:
```suggestion
        var messageTypeName = !string.IsNullOrEmpty(messageTypeHeader) ? messageTypeHeader.Split(',').FirstOrDefault() : default;
```

**Review by SzymonPobiega — APPROVED** [review]
Nitpicking ;)

File: src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs [hunk 3]
```diff
@@ -18,12 +19,14 @@ public ReceiveDiagnosticsBehavior(string queueNameBase, string discriminator)
     public async Task Invoke(IIncomingPhysicalMessageContext context, Func<IIncomingPhysicalMessageContext, Task> next)
     {
         context.MessageHeaders.TryGetValue(Headers.EnclosedMessageTypes, out var messageTypes);
+        var messageTypeHeader = !string.IsNullOrEmpty(messageTypes) ? messageTypes.Split(';').FirstOrDefault() : default;
```
Comment (bbrandt) [inline src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs:22]: Would we ever want to handle a weirdly formatted message type string that starts with a `;`.  Maybe this is too edge and we'd want to fix the thing creating the buggy data instead.

```csharp
using System;
using System.Linq;

namespace HelloWorld
{
	public class Program
	{
		public static void Main(string[] args)
		{
		  var messageTypes = ";ClassName1,unecessary information;ClassName2,unecessary information";
			var messageTypeHeader = !string.IsNullOrEmpty(messageTypes) ? messageTypes.Split(';').FirstOrDefault(s => !string.IsNullOrEmpty(s)) : null;
			var messageTypeName = !string.IsNullOrEmpty(messageTypeHeader) ? messageTypeHeader.Split(',').FirstOrDefault(s => !string.IsNullOrEmpty(s)) : null;
			
			Console.WriteLine(messageTypeName);
		}
	}
}
```

Comment (bbrandt) [reply src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs:22]: @lailabougria Usually, we like to cache these types of string manipulations in a ConcurrentDictionary for performance reasons?

https://github.com/Particular/NServiceBus/blob/fd29208107e147251c10c1663b4c62c0d1e110c4/src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs#L72

https://github.com/Eventellect/SupportRepro/blob/39b6264282688f21737990014e2b5b72faa4b660/src/NSB.Extensions/EnclosedMessageTypesMappingBehavior.cs#L59

**Pull request merged**